src/main/java/org/jenkinsci/plugins/workflow/steps/WaitForConditionStep.java

```diff
@@ -80,7 +116,6 @@
         }
 
         @Override public void onResume() {
-            recurrencePeriod = MIN_RECURRENCE_PERIOD;
```

Comment:
You need to restore `currentRecurrencePeriod` to the minimum here; why was this deleted?